app/src/main/java/org/simple/clinic/registration/pin/RegistrationPinScreen.kt

```diff
@@ -28,6 +30,17 @@ class RegistrationPinScreen(
   @Inject
   lateinit var controller: RegistrationPinScreenController
 
+  private val events by unsafeLazy {
+    Observable
+        .merge(
+            screenCreates(),
+            pinTextChanges(),
+            doneClicks()
+        )
+        .compose(ReportAnalyticsEvents())
+        .share()
```

Comment:
~Can you help me with understanding why are we using `share()` and what it does?~ Are we sharing this because we need to share the events in Controller and Mobius loop as well?

Reply:
Yup, that is right. In the final parts of the refactoring after we remove the controller, we will also stop sharing the stream.